modules/ROOT/pages/advanced-templates.adoc

```diff
@@ -0,0 +1,108 @@
+= Advanced Template plugin
+:navtitle: Advanced Template
+:description: Create and edit complex templates in {productname} for use with {productname}.
+:description_short: Create templates.
+:keywords: plugin, advanced template, templates, advtemplate.
+:pluginname: Advanced Template
+:plugincode: advtemplate
+:plugincategory: premium
+
+include::partial$misc/admon-premium-plugin.adoc[]
+
+include::partial$misc/admon-requires-6.4v.adoc[]
+
+NOTE: The documentation for this page is a work in progress. Further details on the {pluginname} plugin will be added to this page soon.
+
+The {pluginname} Premium plugin enables the creation of complex, interactive templates using {productname}.
+
+They can also be added to a {productname} instance by end-users selecting content in a {productname} document and saving the selection as a template via the {productname} user-interface.
+
+IMPORTANT: This initial {pluginname} release does not support template editing. To update a particular template; insert the template into an editor; make edits; select the, now edited, content; and then create a new template (using, for example, the *Tools > Save as Template…* menu item).
+
+////
+== Interactive example
+
+liveDemo::{plugincode}[]
+////
+
+== Basic setup
+
+To setup the {pluginname} plugin user-interface in the editor
+
+* add `{plugincode}` to the `plugins` option in the editor configuration;
+
+* add `inserttemplate` and `addtemplate` to the `toolbar` option in the editor configuration;
+
+* add each {pluginname} option to the editor configuration;
+
+For example:
+
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your html
+  advtemplate_list: () => {
+    return Promise.resolve([
+      {
+        id: '1',
+        title: 'Resolving tickets',
+        content: '<p>As we have not heard back from you in over a week, so we have gone ahead and resolved your ticket</p>'
+      },
+      {
+        id: '2',
+        title: 'Quick replies',
+        items: [
+          {
+            id: '3',
+            title: 'Message received',
+            content: '<p>Just a quick note to say we have received your message, and will get back to you within 48 hours.</p>'
+          },
+          {
+            id: '4',
+            title: 'Progress update',
+            content: '</p>Just a quick note to let you know we are still working on your case</p>'
+          }
+        ]
+      }
+    ]);
+  }
+});
+----
+
+NOTE: The above example only demonstrates the {pluginname} user interface. An xref:#advtemplate_list[`advtemplate_list`] option has been added, with an included function returning a promise that contains a basic Template structure. This allows the {pluginname} user interface to load and present in a {productname} editor instance. This basic setup example cannot create, edit, or save new {pluginname} items.
```

Comment:
This plugin configured according to the above example will not be able to preview template content

Reply:
> The plugin configured according to the above example will not be able to preview and insert templates' content

Is there any way to set up the promise such that, should someone copy-and-paste the sample into a fiddle, the UI would also preview the selected template and allow the template text be inserted into the TinyMCE editor instance?

Reply:
I’ve pushed an [edit to this sentence](https://github.com/tinymce/tinymce-docs/commit/2cd65418d1645dae6b9dd0f5722a3d39caf0a447) that explicitly acknowledges the limitation noted by @vpyshnenko above.